llmware/parsers.py

```diff
@@ -3066,6 +3066,151 @@ def basic_library_duplicate_check(self, fn):
+
+                for keys, values in mapping_dict.items():
+
+                    if keys == "text":
+                        if mapping_dict["text"] < len(rows):
+                            text = rows[mapping_dict["text"]]
+
+                    if keys == "doc_ID":
+                        if mapping_dict["doc_ID"] < len(rows):
+                            doc_id = rows[mapping_dict["doc_ID"]]
+
+                    if keys == "block_ID":
+                        if mapping_dict["block_ID"] < len(rows):
+                            block_id = rows[mapping_dict["block_ID"]]
+
+                    if keys not in reserved_keys:
+                        if values < len(rows):
+                            metadata.update({keys:rows[values]})
+
+            if text.strip():
+
+                meta = {"author": "", "modified_date": "", "created_date": "", "creator_tool": ""}
+                coords_dict = {"coords_x": 0, "coords_y": 0, "coords_cx": 0, "coords_cy": 0}
+
+                # note: if using SQL-based DB, then will save the metadata as a text string
+                if LLMWareConfig().get_config("collection_db") in ["sqlite", "postgres"]:
+                    metadata = str(metadata)
+
+                new_row_entry = ("text", "custom_csv", (1, 0), total_row_count, "", "", fn,
+                                 text, text, "", "", text, text, "", text, "", "", metadata, "", "")
+
+                #   set attributes custom
+                if doc_id:
+                    try:
+                        self.library.doc_ID = int(doc_id)
+                        added_doc_count += 1
+                    except:
+                        logging.warning(f"update: doc_ID expected to be integer - can not apply custom doc ID "
+                                        f"- {doc_id} - will use default library document increment")
+
+                if block_id:
+                    self.library.block_ID = block_id
+                else:
+                    self.library.block_ID += 1
+
+                #   write row to database
+
+                entry_output = self.add_create_new_record(self.library,
+                                                          new_row_entry,
+                                                          meta,
+                                                          coords_dict,
+                                                          dialog_value="false")
+                added_row_count += 1
+
+            total_row_count += 1
+
+        # update overall library counter at end of parsing
+
+        if len(output) > 0:
+
+            if added_doc_count == 0:
+                added_doc_count += 1
+
+            dummy = self.library.set_incremental_docs_blocks_images(added_docs=added_doc_count,
+                                                                    added_blocks=added_row_count,
+                                                                    added_images=0, added_pages=0)
```

Comment:
:thinking: Since ``dummy`` is not used, please change to
```suggestion
            _ = self.library.set_incremental_docs_blocks_images(added_docs=added_doc_count,
                                                                    added_blocks=added_row_count,
                                                                    added_images=0, added_pages=0)
```